### PR TITLE
feat(figma-utils): add CLI option `selector`

### PR DESCRIPTION
**Pull request by larsrickert**
## Description

- add CLI option `selector` to change the selector used in the `.css` file
- fix: remove selector for SCSS format because this made the variables unusable
- use the `html` selector for our light and dark theme so the selector becomes `html.light` and `html.dark`

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] All changes are documented in the documentation app (folder `apps/docs`)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

**Comment by BoppLi** [comment]
> ## Description
> * add CLI option `selector` to change the selector using in the `.css` file
> * fix: remove selector for SCSS format because this made the variables unusable
> * use the `html` selector for our light and dark theme so the selector becomes `html.light` and `html.dark`
> 
> ## Checklist
> * [x]  The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
> * [x]  All changes are documented in the documentation app (folder `apps/docs`)
> * [x]  If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
> * [x]  A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

suggestion:
`..the selector **used** in the `.css` file`

File: .changeset/sharp-years-lie.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+"@sit-onyx/figma-utils": major
+---
+
+fix(scss): remove `:root` selector
+
+When importing Figma variables with format `SCSS`, the resulting `.scss` file will put the variables
+directly in the root of the file instead of nesting them inside a `:root` selector.
+
+The reason for this is that the variables must be importable / usable into other `.scss` files put nesting
```
Comment (BoppLi) [inline .changeset/sharp-years-lie.md:10]:
```suggestion
The reason for this is that the variables must be importable / usable into other `.scss` files but nesting
```

File: .changeset/sharp-years-lie.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+them instead a selector will make them unusable.
```
Comment (BoppLi) [inline .changeset/sharp-years-lie.md:11]:
```suggestion
them inside a selector will make them unusable.
```

File: packages/figma-utils/src/utils/generate.ts [hunk 3]
```diff
@@ -1,40 +1,24 @@
...
-   * (name) => `var(--${name})`
-   * ```
-   */
-  aliasTransformer: (name: string) => string;
-};
-
 /**
  * Generates the given parsed Figma variables into CSS variables.
  *
+ * @param selector CSS selector to use. The mode name will be added to the selector
```
Comment (BoppLi) [inline packages/figma-utils/src/utils/generate.ts:6]: doesn't JSDoc want to have all params annotated -> if you annotate `selector`, you should do the same for `data`? (either all or none)

File: packages/figma-utils/src/utils/generate.ts [hunk 4]
```diff
@@ -1,40 +1,24 @@
...
+ * if its set to something other than ":root", e.g. "html.dark" is the selector is set to "html"
```
Comment (BoppLi) [inline packages/figma-utils/src/utils/generate.ts:7]:
```suggestion
 * if it is set to something other than ":root", e.g. for the mode named "dark", passing the selector "html" will result in "html.dark"
```

**Review by BoppLi — CHANGES_REQUESTED** [review]
I'll finish reviewing the last file tomorrow.

File: packages/figma-utils/src/utils/generate.ts (hunk 3 above)
Comment (larsrickert) [reply packages/figma-utils/src/utils/generate.ts]: Technically you can only document some props but I added the second one to have it complete

File: packages/figma-utils/src/utils/generate.ts [hunk 5]
```diff
@@ -1,40 +1,25 @@
...
+    const { isAlias, variableName } = isAliasVariable(value);
+    const variableValue = isAlias ? `var(--${variableName})` : value;
+    return `  --${name}: ${variableValue};`;
   });
+
+  selector = selector.trim();
+  if (selector !== ":root") selector += `.${data.modeName}`;
+
+  return `${generateTimestampComment(data.modeName)}
+${selector} {\n${variableContent.join("\n")}\n}\n`;
```
Comment (BoppLi) [inline packages/figma-utils/src/utils/generate.ts:22]: isn't it advised against to change func params directly?
```suggestion
  let fullSelector = selector.trim();
  if (fullSelector !== ":root") fullSelector += `.${data.modeName}`;

  return `${generateTimestampComment(data.modeName)}
${fullSelector} {\n${variableContent.join("\n")}\n}\n`;
```

**Review by BoppLi — APPROVED** [review]

File: packages/figma-utils/src/utils/generate.ts (hunk 5 above)
Comment (larsrickert) [reply packages/figma-utils/src/utils/generate.ts]: Done

**Review by BoppLi — APPROVED** [review]

**Pull request merged**